source/mainnet/smart-contracts/general/develop-contracts.rst

```diff
@@ -288,31 +288,76 @@ Parameters have a size limit of 65kiB. There is no return value size limit (apar
 Working with queries
 --------------------
 
-Queries can be called from smart contracts to query an account balance, contract balance, or the current exchange rates.
+Queries can be called from smart contracts to query an account balance, contract balance,
+or the current exchange rates.
+
+Query an account balance
+~~~~~~~~~~~~~~~~~~~~~~~~
 
 To query account balances, the following are available:
 
 .. code-block:: rust
 
-    let account_balance = host.account_balance(address)?;
-    let available_balance = account_balance.available(); // The amount which can be transferred.
-    let staked_balance = account_balance.staked(); // The staked amount.
-    let locked_balance = account_balance.locked(); // The amount locked in scheduled transfers.
+   // Query the balance of an account.
+   let account_balance = host.account_balance(account_address)?;
+
+Assuming the account exists, this will return all of the public balances of an account,
+which can then be accessed as:
+
+.. code-block:: rust
+
+   // The amount which is available for transfers.
+   let available_balance = account_balance.available();
+   // The staked amount.
+   let staked_balance = account_balance.staked();
+   // The amount locked in scheduled transfers.
+   let locked_balance = account_balance.locked();
+   // The total public balance, i.e. including staked and locked_balance.
+   let total_balance = account_balance.total();
 
-To query contract balance, the following is available:
+.. note::
+
+   When sending a smart contract update transaction, the invoker provides a max energy cost for the execution.
+   CCD equivalent to the max energy cost is then deposited during execution of the contract.
+   Because of this, querying the balance of the invoker will result in the current account balance minus the
+   amount of CCD needed to cover the max energy cost.
```

Comment:
We should also document the semantics of this query with respect to transfers/invoke calls inside a single transaction.

Reply:
That is alluded to for contracts, but not here.

Reply:
I have added to this

Reply:
Thanks.